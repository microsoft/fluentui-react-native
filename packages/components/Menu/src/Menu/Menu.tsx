import React from 'react';

import { stagedComponent } from '@fluentui-react-native/framework';

import type { MenuProps } from './Menu.types';
import { menuName } from './Menu.types';
import { renderFinalMenu } from './renderMenu';
import { useMenu } from './useMenu';
import { useMenuContextValue } from './useMenuContextValue';

export const Menu = stagedComponent((props: MenuProps) => {
  const state = useMenu(props);
  const contextValue = useMenuContextValue(state);

  return (_rest: MenuProps, children: React.ReactNode) => {
    const childrenArray = React.Children.toArray(children) as React.ReactElement[];

    if (__DEV__) {
      if (childrenArray.length !== 2) {
        // eslint-disable-next-line no-console
        console.warn('Menu must contain two children');
      }
    }
    return renderFinalMenu(childrenArray, contextValue, state);
  };
});

Menu.displayName = menuName;

export default Menu;
