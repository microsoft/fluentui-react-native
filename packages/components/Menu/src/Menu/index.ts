export { Menu } from './Menu';
export { menuName } from './Menu.types';
export type { MenuProps, MenuState } from './Menu.types';
export { useMenu } from './useMenu';
export { useMenuContextValue } from './useMenuContextValue';
