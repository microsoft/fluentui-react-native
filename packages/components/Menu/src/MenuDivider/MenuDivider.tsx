/** @jsxRuntime classic */
/** @jsx withSlots */
import { View } from 'react-native';

import type { UseSlots } from '@fluentui-react-native/framework';
import { compose, withSlots } from '@fluentui-react-native/framework';

import { stylingSettings } from './MenuDivider.styling';
import type { MenuDividerProps, MenuDividerType } from './MenuDivider.types';
import { menuDividerName } from './MenuDivider.types';

export const MenuDivider = compose<MenuDividerType>({
  displayName: menuDividerName,
  ...stylingSettings,
  slots: {
    root: View,
  },
  useRender: (userProps: MenuDividerProps, useSlots: UseSlots<MenuDividerType>) => {
    const Slots = useSlots(userProps);

    return (final: MenuDividerProps) => {
      return <Slots.root {...final} />;
    };
  },
});
