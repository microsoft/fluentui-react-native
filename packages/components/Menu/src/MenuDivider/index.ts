export { MenuDivider } from './MenuDivider';
export { menuDividerName } from './MenuDivider.types';
export type { MenuDividerProps, MenuDividerTokens, MenuDividerSlotProps, MenuDividerType, MenuDividerInsetSize } from './MenuDivider.types';
