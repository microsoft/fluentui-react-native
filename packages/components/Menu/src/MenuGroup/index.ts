export { MenuGroup } from './MenuGroup';
export { menuGroupName } from './MenuGroup.types';
export type { MenuGroupProps, MenuGroupTokens, MenuGroupSlotProps, MenuGroupType } from './MenuGroup.types';
