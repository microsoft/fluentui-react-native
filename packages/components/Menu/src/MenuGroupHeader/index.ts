export { MenuGroupHeader } from './MenuGroupHeader';
export { menuGroupHeaderName } from './MenuGroupHeader.types';
export type { MenuGroupHeaderProps, MenuGroupHeaderTokens, MenuGroupHeaderSlotProps, MenuGroupHeaderType } from './MenuGroupHeader.types';
