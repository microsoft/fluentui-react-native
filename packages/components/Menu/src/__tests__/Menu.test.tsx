import * as React from 'react';
import type { AccessibilityActionName } from 'react-native';

import { ButtonV1 as Button } from '@fluentui-react-native/button';
import { checkReRender } from '@fluentui-react-native/test-tools';
import * as renderer from 'react-test-renderer';

import { Menu } from '../Menu/Menu';
import { MenuDivider } from '../MenuDivider/MenuDivider';
import { MenuGroup } from '../MenuGroup';
import { MenuGroupHeader } from '../MenuGroupHeader';
import { MenuItem } from '../MenuItem/MenuItem';
import { MenuItemCheckbox } from '../MenuItemCheckbox/MenuItemCheckbox';
import { MenuItemRadio } from '../MenuItemRadio/MenuItemRadio';
import { MenuList } from '../MenuList/MenuList';
import MenuPopover from '../MenuPopover/MenuPopover';
import MenuTrigger from '../MenuTrigger/MenuTrigger';

describe('Menu component tests', () => {
  it('Menu default', () => {
    const tree = renderer
      .create(
        <Menu>
          <MenuTrigger>
            <Button>Default</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItem>Option 1</MenuItem>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Menu open', () => {
    const tree = renderer
      .create(
        <Menu open>
          <MenuTrigger>
            <Button>Open</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItem>Option 1</MenuItem>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Menu defaultOpen', () => {
    const tree = renderer
      .create(
        <Menu defaultOpen>
          <MenuTrigger>
            <Button>Open</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItem>Option 1</MenuItem>
              <MenuItem disabled>Option 2</MenuItem>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Menu open checkbox and divider', () => {
    const tree = renderer
      .create(
        <Menu open>
          <MenuTrigger>
            <Button>Open</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItemCheckbox name="Option 1">Option 1</MenuItemCheckbox>
              <MenuDivider />
              <MenuItemCheckbox disabled name="Option 2">
                Option 2
              </MenuItemCheckbox>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Menu open radio', () => {
    const tree = renderer
      .create(
        <Menu open>
          <MenuTrigger>
            <Button>Open</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItemRadio name="Option 1">Option 1</MenuItemRadio>
              <MenuItemRadio name="Option 2">Option 2</MenuItemRadio>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Menu open checkbox defaultChecked', () => {
    const tree = renderer
      .create(
        <Menu open defaultChecked={['Option 1']}>
          <MenuTrigger>
            <Button>Open</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItemCheckbox name="Option 1">Option 1</MenuItemCheckbox>
              <MenuDivider />
              <MenuItemCheckbox name="Option 2">Option 2</MenuItemCheckbox>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Menu open checkbox checked', () => {
    const tree = renderer
      .create(
        <Menu open checked={['Option 1']}>
          <MenuTrigger>
            <Button>Open</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItemCheckbox name="Option 1">Option 1</MenuItemCheckbox>
              <MenuDivider />
              <MenuItemCheckbox name="Option 2">Option 2</MenuItemCheckbox>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Menu submenu', () => {
    const tree = renderer
      .create(
        <Menu open>
          <MenuTrigger>
            <Button>Default</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItem>Option 1</MenuItem>
              <Menu>
                <MenuTrigger>
                  <MenuItem>Option 2</MenuItem>
                </MenuTrigger>
                <MenuPopover>
                  <MenuList>
                    <MenuItem>Option 1</MenuItem>
                  </MenuList>
                </MenuPopover>
              </Menu>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Note the capital T the "Tip" (in the snapshot)
  // It is intentional as it matches the same prop in NetUI
  it('Menu alwaysShowToolTip', () => {
    const tree = renderer
      .create(
        <Menu>
          <MenuTrigger>
            <Button tooltip="A Tooltip">Button</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItem>Test</MenuItem>
              <MenuItem tooltip="Second">With Tooltip</MenuItem>
              <MenuItemRadio name="Radio" tooltip="Third">
                Radio Tooltip
              </MenuItemRadio>
              <MenuItemCheckbox name="Check" tooltip="Fourth">
                Checkbox Tooltip
              </MenuItemCheckbox>
            </MenuList>
          </MenuPopover>
        </Menu>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('Menu open menu group and menu header', () => {
  const tree = renderer
    .create(
      <Menu open>
        <MenuTrigger>
          <Button>Open</Button>
        </MenuTrigger>
        <MenuPopover>
          <MenuList>
            <MenuGroup>
              <MenuGroupHeader>Header 1</MenuGroupHeader>
              <MenuItem>Option 1</MenuItem>
            </MenuGroup>
            <MenuDivider></MenuDivider>
            <MenuGroup>
              <MenuGroupHeader>Header 2</MenuGroupHeader>
              <MenuItem>Option 1</MenuItem>
            </MenuGroup>
          </MenuList>
        </MenuPopover>
      </Menu>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Menu rerender tests', () => {
  it('Menu re-renders correctly', () => {
    checkReRender(
      () => (
        <Menu open>
          <MenuTrigger>
            <Button>Rerender twice</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItem>Option 1</MenuItem>
            </MenuList>
          </MenuPopover>
        </Menu>
      ),
      3,
    );
  });

  it('Menu re-renders correctly with style', () => {
    const style = { backgroundColor: 'black' };
    checkReRender(
      () => (
        <Menu>
          <MenuTrigger>
            <Button style={style}>Rerender twice</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItem>Option 1</MenuItem>
            </MenuList>
          </MenuPopover>
        </Menu>
      ),
      3,
    );
  });

  it('Menu re-renders correctly with accessibilityAction', () => {
    const action = [{ name: 'Expand' as AccessibilityActionName }];
    checkReRender(
      () => (
        <Menu>
          <MenuTrigger>
            <Button>Rerender twice</Button>
          </MenuTrigger>
          <MenuPopover>
            <MenuList>
              <MenuItem accessibilityActions={action}>Option 1</MenuItem>
            </MenuList>
          </MenuPopover>
        </Menu>
      ),
      3,
    );
  });
});
