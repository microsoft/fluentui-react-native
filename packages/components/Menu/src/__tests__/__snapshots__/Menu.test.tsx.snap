// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component tests Menu alwaysShowToolTip 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": false,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  alwaysShowToolTip={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
  tooltip="A Tooltip"
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Button
  </Text>
</View>
`;

exports[`Menu component tests Menu default 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Default"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": false,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Default
  </Text>
</View>
`;

exports[`Menu component tests Menu defaultOpen 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Open"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": true,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Open
  </Text>
</View>
`;

exports[`Menu component tests Menu open 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Open"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": true,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Open
  </Text>
</View>
`;

exports[`Menu component tests Menu open checkbox and divider 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Open"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": true,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Open
  </Text>
</View>
`;

exports[`Menu component tests Menu open checkbox checked 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Open"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": true,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Open
  </Text>
</View>
`;

exports[`Menu component tests Menu open checkbox defaultChecked 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Open"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": true,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Open
  </Text>
</View>
`;

exports[`Menu component tests Menu open radio 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Open"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": true,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Open
  </Text>
</View>
`;

exports[`Menu component tests Menu submenu 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Default"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": true,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Default
  </Text>
</View>
`;

exports[`Menu open menu group and menu header 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Expand",
      },
      {
        "name": "Collapse",
      },
    ]
  }
  accessibilityLabel="Open"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": true,
      "multiselectable": undefined,
      "required": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  enableFocusRing={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onAccessibilityTap={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#d1d1d1",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 24,
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 7,
      "width": undefined,
    }
  }
>
  <Text
    accessible={false}
    ellipsizeMode="tail"
    numberOfLines={0}
    onAccessibilityTap={[Function]}
    style={
      {
        "color": "#242424",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 1,
        "marginEnd": -2,
        "marginStart": 0,
        "marginTop": -1,
      }
    }
  >
    Open
  </Text>
</View>
`;
